lib/rbac/filterer.rb

```diff
@@ -470,12 +470,8 @@ def scope_targets(klass, scope, rbac_filters, user, miq_group)
 
         filtered_ids = calc_filtered_ids(associated_class, rbac_filters, user, miq_group, scope_tenant_filter)
         scope_by_parent_ids(associated_class, scope, filtered_ids)
-      elsif klass == User && user.try!(:self_service?)
-        # Self service users searching for users only see themselves
-        scope.where(:id => user.id)
-      elsif klass == MiqGroup && miq_group.try!(:self_service?)
-        # Self Service users searching for groups only see their group
-        scope.where(:id => miq_group.id)
+      elsif [User, MiqGroup].include?(klass) && (miq_group || user).try!(:self_service?)
```

Comment:
Interestingly, we can do this. Since the miq_group getting here should be always equal to user.current_group (if both supplied). And the self_service? method get's routed to the current_group.

Net result is the same. Even though, I keep scratching my chin at this. :)